dashboard/app/models/concerns/text_to_speech.rb

```diff
@@ -37,38 +37,20 @@ def image(link, title, alt_text)
 module TextToSpeech
   extend ActiveSupport::Concern
 
-  VOICES = {
-    'en-US': {
-      VOICE: 'sharon22k',
-      SPEED: 180,
-      SHAPE: 100
-    },
-    'es-ES': {
-      VOICE: 'ines22k',
-      SPEED: 180,
-      SHAPE: 100,
-    },
-    'es-MX': {
-      VOICE: 'rosa22k',
-      SPEED: 180,
-      SHAPE: 100,
-    },
-    'it-IT': {
-      VOICE: 'vittorio22k',
-      SPEED: 180,
-      SHAPE: 100,
-    },
-    'pt-BR': {
-      VOICE: 'marcia22k',
-      SPEED: 180,
-      SHAPE: 100,
-    }
-  }.freeze
-
-  # TODO: this concern actually depends on the SerializedProperties
-  # concern ... I'm not sure how best to deal with that.
+  # Pull the VOICES out of the SharedConstants (updating the locale keys to
+  # match the expectation of the I18n gem since these locale keys are using
+  # the JavaScript i18n expectations)
+  VOICES = SharedConstants::VOICES.transform_keys do |locale|
+    lang, country = locale.to_s.split('_')
+    :"#{lang}-#{country.upcase}"
```

Comment:
What if the locale does not have an underscore? My understanding is that the system's locale can be a two-letter code as well. Maybe using a `gsub` statement could be safer since it would return the same string if no matches are found.

Reply:
I see this is coming from shared constants:
https://github.com/code-dot-org/code-dot-org/blob/ab089c08ff56c57817d4523ee1ba7803e1be70f4/lib/cdo/shared_constants.rb#L655-L681

Reply:
is there a reason to switch the underscore to a dash?

Reply:
Can you recommend a `gsub` that would convert the JavaScript locale code `en_us` to the style the Ruby localization system expects (`en-US`)?

Reply:
We could do something like:
```
def replace_underscore(string)
  captures = string.tr('_', '-').match(/([A-Za-z]{2})(-[A-Za-z]*)?/).captures
  if captures[1]
    captures[0] + captures[1].upcase
  else
    captures[0]
  end
end
```
But this would be over-engineering. Your solution looks great if we are confident we are expecting the two-letter locale.

Reply:
I can handle this by adding that `return` line to ignore the locale if it does not have an underscore:

```
VOICES = SharedConstants::VOICES.transform_keys do |locale|
  return locale unless locale.to_s.include?('_')
  lang, country = locale.to_s.split('_')
  :"#{lang}-#{country.upcase}"
end.freeze
```

They all should have them, though.